ludwig/automl/automl.py

```diff
@@ -278,17 +280,25 @@ def _model_select(
         # text heuristics
         for input_feature in base_config["input_features"]:
             # default text encoder is bert
-            if input_feature["type"] == TEXT:
+            if input_feature[TYPE] == TEXT:
                 model_category = TEXT
-                input_feature["encoder"] = AUTOML_DEFAULT_TEXT_ENCODER
-                base_config = merge_dict(base_config, default_configs[TEXT][AUTOML_DEFAULT_TEXT_ENCODER])
+                if ENCODER in input_feature:
+                    input_feature[ENCODER][TYPE] = AUTOML_DEFAULT_TEXT_ENCODER
+                else:
+                    input_feature[ENCODER] = {TYPE: AUTOML_DEFAULT_TEXT_ENCODER}
                 base_config[HYPEROPT]["executor"]["num_samples"] = 5  # set for small hyperparameter search space
 
             # TODO (ASN): add image heuristics
-            if input_feature["type"] == IMAGE:
+            if input_feature[TYPE] == IMAGE:
                 model_category = IMAGE
-                input_feature["encoder"] = AUTOML_DEFAULT_IMAGE_ENCODER
-                base_config = merge_dict(base_config, default_configs["combiner"]["concat"])
+                if ENCODER in input_feature:
+                    input_feature[ENCODER][TYPE] = AUTOML_DEFAULT_IMAGE_ENCODER
+                else:
+                    input_feature[ENCODER] = {TYPE: AUTOML_DEFAULT_IMAGE_ENCODER}
+
+        # Needs to be outside for loop because merge dict creates deep copy - this prevents image section from setting
+        base_config = merge_dict(base_config, default_configs[TEXT][AUTOML_DEFAULT_TEXT_ENCODER])
```

Comment:
This is good callout @martindavis in this case though, default_configs is not the usual defaults section of the config that we're used to. It's actually a specific dictionary for automl with a only 3 sections where the automl defaults for text are sitting under `text` in this dict and the image defaults are under `combiner` for some reason. I added coverage for this file so it is working as expected, but think that it could probably do with a bit a refactor down the line so it makes a little more sense. For now I believe that is out of scope though.